packages/anu-vue/src/composables/useZIndex.ts

```diff
@@ -0,0 +1,35 @@
+import type { InjectionKey, Ref } from 'vue'
+import { computed, inject, ref, unref } from 'vue'
+import { isNumber } from '@vueuse/core'
+
+// TODO add Config Provider Component
+export const zIndexContextKey: InjectionKey<Ref<number | undefined>> = Symbol('zIndexContextKey')
+
+const zIndexCounter = ref(0)
+export const defaultBaseZIndex = 2000 as const
```

Comment:
How about, If we pass this as config to [`useAnu`](https://github.com/jd-solanki/anu/blob/45b0ac101343315cc1506654f30cea9d0f621643/packages/anu-vue/src/composables/useAnu.ts#L16) with a default value of 2000 as per your code?

Reply:
This looks like a pretty good idea, but when we consider potentially needing more global-level configurations later on, wouldn't this make the `useAnu` function somewhat overly complex?

Reply:
Yes it can but I had a thought that what will be best where we allow users to configure z-index then?

ATM, we are configuring themes in useAnu initially here: https://github.com/jd-solanki/anu/blob/main/packages/anu-vue/src/plugin.ts#L77-L80

Later, users can use `useAnu` composable to get the configured and runtime options like theme (only ATM).

Maybe we can move theme only composable to seperate composable like vuetify, `useTheme`. (_I don't have implementation idea yet_)

What's your thoughts?

Reply:
> _P.S. In future, we will provide defaults provider like `el-config-provider` in future._

Reply:
Do you mean the Discord app? My username is Icet#3793.

I'm also pondering which approach to choose, and it seems there are two viable options at the moment:
- `ConfigProvider`
```vue
<template>
  <a-config-provider :zIndex="zIndex">
    <app />
  </a-config-provider>
</template>

<script setup>
const zIndex = ref(1000)
<script/>
```
- `useAnuConfig`
```vue
<template>
   <app />
</template>

<script setup>
const zIndex = ref(1000)
useAnuConfig({ zIndex })
<script/>
```
We may need more time to make a trade-off.

Reply:
Alright, I got the idea. We are already using useAnu via one-time configuration thanks to `createCustomGlobalState`. 

originally it's [`createGlobalState`](https://vueuse.org/shared/createGlobalState/#createglobalstate) - we need to replace this composable with vueuse's composable, however I'm getting some type error I guess which I haven't looked into yet, if you got time feel free to remove `createCustomGlobalState` and use official composable.

With this one time configuration, we can accept base z-index config via anu config `app.use(anu, {})` and initialize useZIndex with baseZIndex. I hope you got the idea. This is same thing we did with `useAnu`.

Let me know your thoughts on this, Awaiting your response.

BTW, I couldn't find you on the discord. Are you on the Anu's [channel](https://discord.gg/8MTQuasmZf)?

Reply:
I have pushed new commit for useAnu improvement: https://github.com/jd-solanki/anu/commit/0bc880d992d407890cea6bdf551778e45f6bcf5d

Now, It might be clear, What I was saying. We will create useZIndex via `createGlobalState` composable like [this](https://github.com/jd-solanki/anu/blob/main/packages/anu-vue/src/composables/useAnu.ts#L16) and initialize it for the first time via user config like [this](https://github.com/jd-solanki/anu/blob/main/packages/anu-vue/src/plugin.ts#L77-L80). To configure, base z-index user can just pass it as config property like [this](https://github.com/jd-solanki/anu/blob/main/docs/guide/features/theme.md?plain=1#L21).

This will allow us to create as many composable as we want with clear seperate of what each composable do.

For example, we already have `useZIndex` thanks to you. We can also convert `useAnu` composable to `useTheme` that solely handles theme related state.

According to me this is the best possible way to configure Anu now.

**Overview:**
- Pass user config via `app.use(anu, { ... })`
- We initialize as many composable as we want for clear seperation
- (Additionally) we will be able to use this composable in features like defaults provider later

Reply:
Sorry for taking a while to reply.

I think this method is great. Using configuration data at the global scope level gives us a lot of flexibility in providing more configuration options. It also solves the problem I was having about how to handle priority when there is a nested provider.

Reply:
Hey @IcetCode 

Sorry I was busy lately. Just wanted to know, Are you going to update the PR or I should go ahead?

Reply:
Hey @jd-solanki 

I've been swamped with work too. I'll update the PR tomorrow for `app.use(anu, { zIndex })` and `config-provider`. What do you think about this?

Reply:
No worries, I had time so I thought I can work on it if you don't have time.

I will wait for the update 😇

Thanks